### PR TITLE
made open file only analyzer to run properly for fix all and suppression.

### PR DESCRIPTION
**Pull request by heejaechang**
**Customer scenario**

User apply "fix all" on a solution or invoke "add suppression" from error list for an error, but VS doesn't do anything.

**Bugs this fixes:**

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/399140

**Workarounds, if any**

use fix/suppression on open file. but it means user can not run these in bulk.

**Risk**

Low. 

code change is not that complicate. most of changes are plumbing to flow down a flag. actual fix is running those in proc only analyzers in proc at the end rather than either filter them out or run in OOP.

**Performance impact**

it now runs fixers it used to not due to this bug. but that is necessary cost to provide the fix.

**Is this a regression from a previous update?**

Yes.

when OpenFileOnly analyzer concept is added to certain analyzers, those analyzer got regressed.

we had test to cover fix all, but test always used Open Files that is why this bug didnt get caught. changed test to not use open files for fix all case.

**Root cause analysis:**

I basically missed the case where OpenFileOnly analyzers can be used for whole project if "Fix All" is used. so didn't add support for OpenFileOnly analyzers for "Fix All" (error list suppression basically use same code path)

**How was the bug found?**

customer report, dogfooding.

### CODE REVIEW

**Comment by Pilchie** [comment]
@sharwell can you review, and build locally and buddy test against some of the Fix All bugs you have?

**Comment by sharwell** [comment]
:memo: (note to self) See #17183 for test cases.

**Comment by heejaechang** [comment]
@dotnet/roslyn-analysis can you take a look? @mavasani can you take a look?

File: src/VisualStudio/Core/Test.Next/Services/VisualStudioDiagnosticAnalyzerExecutorTests.cs [hunk 1]
```diff
@@ -157,7 +157,10 @@ void Method()
                 var project = workspace.CurrentSolution.Projects.First();
 
                 var executor = (ICodeAnalysisDiagnosticAnalyzerExecutor)new DiagnosticAnalyzerExecutor(new MyUpdateSource(workspace)).CreateService(workspace.Services);
-                var analyzerDriver = (await project.GetCompilationAsync()).WithAnalyzers(analyzerReference.GetAnalyzers(project.Language).Where(a => a.GetType() == analyzerType).ToImmutableArray());
```
Comment (mavasani) [inline src/VisualStudio/Core/Test.Next/Services/VisualStudioDiagnosticAnalyzerExecutorTests.cs]: Where do we filter out disabled/suppressed analyzers?

Comment (mavasani) [inline src/VisualStudio/Core/Test.Next/Services/VisualStudioDiagnosticAnalyzerExecutorTests.cs]: Ah, this is test code.

**Comment by heejaechang** [comment]
@sharwell can you let me know whether this address your issue?

**Comment by heejaechang** [comment]
any other one to take a look? @sharwell done with testing? all unit tests passed and my manual testing all worked so I would like to check this in.

**Comment by sharwell** [comment]
Am taking a look! 😄

**Comment by heejaechang** [comment]
ping?

**Comment by heejaechang** [comment]
@sharwell this change and this (https://github.com/dotnet/roslyn/pull/19637) combined might solve more fix all issues.

**Comment by sharwell** [comment]
I'm having trouble running a local build right now, which is delaying me. Unrelated to your changes.

**Comment by sharwell** [comment]
## Validation Results (in progress...)

### Case reported by @AlexanderTaeschner in #17183 

Details:

* Full solution analysis enabled
* Create a new legacy project, **do not close it**
* Set all prefer type to var, suggestion in Tools &rarr; Options
* Fix All in Solution on `int a = 3;`

Result:

* Reliably failed prior to this change
* *[in progress]* after this change

**Comment by sharwell** [comment]
@heejaechang I finished my validation passes against the known issues believed to be caused by this. I was not able to identify any new regressions in my testing, however it was limited to behavior very similar to the know problematic cases. Please see the results above.

**Comment by heejaechang** [comment]
@sharwell thank you Sam. the last issue probably different issue where editconfig is not flown to engine properly. we can take a look at those next.

and sure, I would appreciate if you can code review.

thank you

**Comment by heejaechang** [comment]
tagging @MattGertz as well to get approval for 15.3

**Pull request merged**